datasets/arabic_pos_dialect/arabic_pos_dialect.py

```diff
@@ -0,0 +1,129 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO: Add a description here."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import datasets
+
+
+_CITATION = """\
+@InProceedings{DARWISH18.562,  author = {Kareem Darwish ,Hamdy Mubarak ,Ahmed Abdelali ,Mohamed Eldesouki ,Younes Samih ,Randah Alharbi ,Mohammed Attia ,Walid Magdy and Laura Kallmeyer},
+title = {Multi-Dialect Arabic POS Tagging: A CRF Approach},
+booktitle = {Proceedings of the Eleventh International Conference on Language Resources and Evaluation (LREC 2018)},
+year = {2018},
+month = {may},
+date = {7-12},
+location = {Miyazaki, Japan},
+editor = {Nicoletta Calzolari (Conference chair) and Khalid Choukri and Christopher Cieri and Thierry Declerck and Sara Goggi and Koiti Hasida and Hitoshi Isahara and Bente Maegaard and Joseph Mariani and Hélène Mazo and Asuncion Moreno and Jan Odijk and Stelios Piperidis and Takenobu Tokunaga},
+publisher = {European Language Resources Association (ELRA)},
+address = {Paris, France},
+isbn = {979-10-95546-00-9},
+language = {english}
+}
+"""
+
+_DESCRIPTION = """\
+The Dialectal Arabic Datasets contain four dialects of Arabic, Etyptian (EGY), Levantine (LEV), Gulf (GLF), and Maghrebi (MGR). Each dataset consists of a set of 350 manually segmented and POS tagged tweets.
+"""
+
+_URL = "https://github.com/qcri/dialectal_arabic_resources/raw/master/"
+_DIALECTS = ["egy", "lev", "glf", "mgr"]
+
+
+class ArabicPosDialectConfig(datasets.BuilderConfig):
+    """ BuilderConfig for ArabicPosDialect"""
+
+    def __init__(self, dialect=None, **kwargs):
+        """
+
+        Args:
+            dialect: the 3-letter code string of the dialect set that will be used.
+            Code should be one of the following: egy, lev, glf, mgr.
+            **kwargs: keyword arguments forwarded to super.
+        """
+        super(ArabicPosDialectConfig, self).__init__(**kwargs)
+        assert dialect in _DIALECTS, ("Invalid dialect code: %s", dialect)
+        self.dialect = dialect
+
+
+class ArabicPosDialect(datasets.GeneratorBasedBuilder):
+    """POS-tagged Arabic tweets in four major dialects."""
+
+    VERSION = datasets.Version("1.1.0")
+
```

Comment:
```suggestion
    BUILDER_CONFIG_CLASS = ArabicPosDialectConfig
```
we need that to specify that we use a custom config